crates/bevy_ecs/src/archetype.rs

```diff
@@ -411,25 +510,29 @@ impl Default for Archetypes {
         archetypes.get_id_or_insert(TableId::empty(), Vec::new(), Vec::new());
         archetypes
     }
-}
 
-impl Archetypes {
     #[inline]
     pub fn generation(&self) -> ArchetypeGeneration {
         ArchetypeGeneration(self.archetypes.len())
     }
 
+    /// Fetches the total number of [`Archetype`]s within the world.
```

Comment:
I kinda want to add something about how empty archetypes don't get cleaned up here, but I'm not sure we'll remember to come back and fix it.

Reply:
Added a comment about this at the module level docs.